apps/passport-client/components/shared/PCDCard.tsx

```diff
@@ -1,18 +1,21 @@
 import { PCD } from "@pcd/pcd-types";
-import { useCallback, useContext, useMemo } from "react";
+import React, { useCallback, useContext, useMemo } from "react";
 import styled from "styled-components";
 import { appConfig } from "../../src/appConfig";
-import { DispatchContext } from "../../src/dispatch";
+import { usePCDCollection, useSelf } from "../../src/appHooks";
+import { StateContext } from "../../src/dispatch";
 import { usePackage } from "../../src/usePackage";
 import { getVisitorStatus, VisitorStatus } from "../../src/user";
 import { Button, H4, Spacer, TextCenter } from "../core";
 import { MainIdentityCard } from "./MainIdentityCard";
 
+export const PCDCard = React.memo(PCDCardImpl);
```

Comment:
one of the main optimizations of this PR - wrap a bunch of components in `React.memo` so that they don't re-render unless their props changed. It's important to be careful when making such optimizations.